plugin/parser/differ/mysql/differ.go

```diff
@@ -6,25 +6,72 @@ import (
 
 	"github.com/pingcap/tidb/parser/ast"
 	"github.com/pingcap/tidb/parser/format"
+	"github.com/pingcap/tidb/parser/model"
 )
 
+type columnKey struct {
+	tableName  string
+	columnName string
+}
+
 // SchemaDiff returns the schema diff.
 func SchemaDiff(old, new []ast.StmtNode) (string, error) {
-	oldTableMap := make(map[string]ast.Node)
-	tableDiff := make([]ast.Node, 0)
+	oldTableMap := make(map[string]*ast.CreateTableStmt)
+	var diff []ast.Node
 	for _, node := range old {
-		if stmt, ok := node.(*ast.CreateTableStmt); ok {
+		switch stmt := node.(type) {
+		case *ast.CreateTableStmt:
 			tableName := stmt.Table.Name.String()
-			oldTableMap[tableName] = node
+			oldTableMap[tableName] = stmt
+		default:
 		}
 	}
+
+	newTableMap := make(map[string]*ast.CreateTableStmt)
 	for _, node := range new {
-		if stmt, ok := node.(*ast.CreateTableStmt); ok {
-			if _, ok := oldTableMap[stmt.Table.Name.String()]; !ok {
-				stmt.IfNotExists = true
-				tableDiff = append(tableDiff, node)
+		switch stmt := node.(type) {
+		case *ast.CreateTableStmt:
+			tableName := stmt.Table.Name.String()
+			newTableMap[tableName] = stmt
+		default:
+		}
+	}
+
+	for tableName, newStmt := range newTableMap {
+		oldStmt, ok := oldTableMap[tableName]
+		if !ok {
+			newStmt.IfNotExists = true
```

Comment:
Consider making a copy of the object instead of modifying the original data structure content.